### PR TITLE
Convert split proofs for contract args

### PR DESCRIPTION
**Pull request by katat**
The following proof formats are converted in json to align with the contract interfaces.
- main proof
- merkle statements
- fri merkle statements

**Logic for merkle statements**
```
def build_merkle_proof_tx_args(input_json) -> Tuple[List[int], List[int], int, int]:
    """
    Create a tx for merkle proof verification.
    """
    statement_contract_input = MerkleStatement.load(data=input_json)
    return statement_contract_args(statement_contract_input)


def statement_contract_args(
    statement_contract_input: MerkleStatement,
) -> Tuple[List[int], List[int], int, int]:
    return (
        statement_contract_input.proof,
        statement_contract_input.merkle_queue,
        statement_contract_input.merkle_height,
        statement_contract_input.
```

**Logic for FRI merkle statements**
```
def build_fri_merkle_proof_tx_args(statement_contract_input: dict):
    """
    Create a tx for fri-merkle proof verification.
    """
    statement_contract_args = (
        statement_contract_input["proof"],
        list(statement_contract_input["input_interleaved"]) + [0],
        statement_contract_input["evaluation_point"],
        statement_contract_input["fri_step_size"],
        statement_contract_input["expected_root"],
    )
    return statement_contract_args
```

**Logic for main proof conversion**
```
def proof_hex2int_list(proof_hex: str) -> List[int]:
    """
    Gets a hex string integer and returns it as a 256bits padded list of integer.
    This conversion is what's needed in order to send a binary proof
    (represented as a HexString) into an EVM deployed verifier.
    """
    assert not proof_hex.startswith("0x"), "proof_hex must be without 0x prefix"
    chunk_size = 64
    padded_proof = proof_hex.ljust(len(proof_hex) + (-len(proof_hex)) % chunk_size, "0")
    return [
        int(chunk, 16)
        for chunk in (
            padded_proof[i : i + chunk_size] for i in range(0, len(padded_proof), chunk_size)
        )
    ]
```

### CODE REVIEW

File: src/annotation_parser.rs [hunk 1]
```diff
@@ -82,10 +83,13 @@ struct FriMerklesOriginal {
...
 /// [SplitProofs] maps the split proof json file which contains the main proof and the merkle statements
 pub struct SplitProofs {
-    #[serde_as(as = "serde_with::hex::Hex")]
-    pub main_proof: Vec<u8>,
-    pub merkle_statements: HashMap<String, MerkleStatement>,
-    pub fri_merkle_statements: Vec<FRIMerkleStatement>,
+    #[serde(
+        serialize_with = "serialize_vec_u256_as_number",
+        deserialize_with = "deserialize_vec_u256_as_number"
+    )]
```
[the remark this answers is not in the document]
Comment (mimoo) [reply src/annotation_parser.rs:89]: or perhaps we should create our own type somewhere? These serialize helpers feel a bit dirty

File: src/annotation_parser.rs [hunk 2]
```diff
@@ -768,9 +772,45 @@ pub fn split_fri_merkle_statements(proof_json: AnnotatedProof) -> Result<SplitPr
...
+/// This conversion is what's needed in order to send a binary proof
+/// into an EVM deployed verifier.
+fn proof_hex2int_list(proof: Vec<u8>) -> Vec<U256> {
+    let chunk_size = 32; // U256 is 32 bytes (256 bits)
+    let mut padded_proof = proof;
+
+    // Pad the vector with zeros until its length is a multiple of chunk_size
+    while padded_proof.len() % chunk_size != 0 {
+        padded_proof.push(0);
+    }
```
Comment (mimoo) [inline src/annotation_parser.rs:806]: maybe cleaner?

```rust
let padded_proof = proof.iter().cloned().chain(std::iter::repeat(0).take(chunk_size - (proof.len() % chunk_size))).collect_vec();
```

Comment (mimoo) [reply src/annotation_parser.rs:806]: or

```rust
let mut padded_proof = proof;
padded_proof.resize(proof.len() + (chunk_size - proof.len() % chunk_size), 0);
```

**Review by mimoo — APPROVED** [review]

File: src/annotation_parser.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (katat) [reply src/annotation_parser.rs:89]: At the moment, these serializers are for json output. Based on the corrected requirement, they are intended to send to contract interfaces, to which solidity ABI serialize/deserialize should be utilized instead.

In other words, it'd better to use ethers library to do the job serializing job when submitting the transactions to the contracts. 

Then these serializers/deserializer, which are more for testing purposes at the moment (for communicating the formats), can be removed.

**Pull request merged**